homeassistant/components/bmw_connected_drive/sensor.py

```diff
@@ -380,111 +133,10 @@ async def async_setup_entry(
                         BMWConnectedDriveSensor(
                             account, vehicle, description, unit_system
                         )
-                        for attribute_name in vehicle.drive_train_attributes
-                        if attribute_name in vehicle.available_attributes
-                        and (description := SENSOR_TYPES.get(attribute_name))
+                        for attribute_name in vehicle.available_attributes
+                        if (description := SENSOR_TYPES.get(attribute_name))
                     ]
                 )
-            if service == SERVICE_LAST_TRIP:
```

Comment:
We can probably remove the pylint disable above since we remove a lot of nesting here.

Reply:
Yes. Pylint succeeds when removed.